source/Politics.cpp

```diff
@@ -248,12 +248,18 @@ string Politics::Fine(PlayerInfo &player, const Government *gov, int scan, const
 		}
 		if(!scan || (scan & ShipEvent::SCAN_OUTFITS))
 		{
+			int64_t fine = 0;
 			for(const auto &it : ship->Outfits())
 				if(it.second)
 				{
-					int64_t fine = it.first->Get("illegal");
-					if(it.first->Get("atrocity") > 0.)
+					if(it.first->Get("atrocity") > 0)
 						fine = -1;
+					else
+					{
+						for(int i = 1; i <= it.second; i++)
```

Comment:
i would ensure you are fined for the minimum of 1 outfit by incrementing the fine once per illegal outift type, and then looping from the 2nd index through the remaining.

It's also rather likely that this is a 0-base indexed list, and you should probably be using `++i`
A good way to test would be to disable the scan interference check and make sure you get fined 235M and not 234M.